src/passes/transform/fold_constants.cpp

```diff
@@ -474,6 +569,45 @@ namespace CoreIR {
             
         }
         
+      } else if (getQualifiedOpName(*(inst)) == "coreir.slice") {
+        // Select* in = inst->sel("in");
+
+        // vector<Select*> in0Values = getSignalValues(in);
+        // maybe<BitVec> sigValue0 = getSignalBitVec(in0Values);
+
+        // if (sigValue0.has_value()) {
+
+        //   BitVec sigVal0 = sigValue0.get_value();
+
+        //   // TODO: Check for binary values
+        //   BitVec res = ~sigVal0;
+
+        //   auto newConst =
+        //     def->addInstance(inst->toString() + "_const_replacement",
+        //                      "coreir.const",
+        //                      {{"width", Const::make(c, res.bitLength())}},
+        //                      {{"value", Const::make(c, res)}});
+
+        //   auto recInstances = getReceiverSelects(inst);
+        //   for (auto elem : recInstances) {
+        //     auto src = extractSource(elem);
+        //     if (isa<Instance>(src)) {
+        //       toConsider.insert(cast<Instance>(src));
+        //     }
+        //   }
+            
+        //   Instance* instPT = addPassthrough(inst, "_inline_andr_PT");
+        //   Select* replacement = newConst->sel("out");
+
+        //   def->removeInstance(inst);
+        //   def->connect(replacement,
+        //                instPT->sel("in")->sel("out"));
+        //   inlineInstance(instPT);
+
+        // }
+
+        cout << "Not folding slice" << endl;
```

Comment:
Similarly, if this is an error, use ASSERT. if you want debug statements, you can use:

LOG(INFO) << "comments";

Check out src/binary/coreir.cpp for an example